modules/ROOT/pages/dataweave-extension-plugin.adoc

```diff
@@ -0,0 +1,397 @@
+= Installing and Using the DataWeave 2.0 Extension
+
+Install and use DataWeave 2.0 extension to Visual Studio Code (DataWeave extension) to develop, preview, test, and deploy custom mappings and modules into libraries in a DataWeave project. The extension also enables you to preview transformations similar to those you perform in the https://developer.mulesoft.com/learn/dataweave/[DataWeave Playground].
+
+== Before You Begin
+
+Before installing the DataWeave 2.0 extension to Visual Studio Code, you must:
+
+* Install AdoptOpenJDK (Java 8 minimum)
+* Install Apache Maven (3.6.3 minimum)
+* Install Microsoft Visual Studio Code for your specific operating system
+
+== Install the DataWeave 2.0 Extension
+
+To install the extension:
+
+. Open Visual Studio Code.
+. In the *EXTENSIONS* tab, type `dataweave` to search the extension in the marketplace.
+. Select *DataWeave 2.0* from the search results.
+. Click *Install*.
++
+image::dataweave-extension-plugin-install.png[DataWeave 2.0 extension installation]
+
+[start=5]
+. After the installation finishes, restart Visual Studio Code.
+
+
+//image::animations/install_plugin.gif[Installation Steps]
+// ![Installation Steps](animations/install_plugin.gif)
+
+== Developing DataWeave Libraries
+
+A DataWeave library is a reusable package of DataWeave module and mapping files, and resource files, such as JSON, XML, and CSV. To develop DataWeave libraries, first create a new DataWeave project in Visual Studio Code where you then create, preview, and test custom mappings and modules.
+
+If you want to try out your mapping without creating a DataWeave project, you can either open your `.dwl` mapping file directly in Visual Studio Code or create a new one and use <<sample-data,sample data>> and <<scenarios,scenarios>>.
+
+=== Understanding the DataWeave Project Structure
+
+DataWeave projects use Apache Maven to build and declare project dependencies. The following diagram shows the DataWeave project structure:
+
+```
+├── src
+│   ├── main
+│   │   ├── dw
+│   │   │   ├── MyMapping.dwl  //Put your DataWeave mapping file here
+|   |   |   └── MyModule.dwl   //Put your DataWeave module file here
+│   │   └── resources
+│   └── test
+│       ├── dw
+│       │   └── Test.dwl  //Unit tests
+│       │   └── MyMapping
+│       │       └── default
+│       │           └── inputs
+│       │               └── payload.json //[Sample data](#sample-data) for your mappings
+│       ├── resources
+├── pom.xml
+```
```

Comment:
```suggestion
├── src
│   ├── main
│   │   ├── dw
│   │   │   ├── MyMapping.dwl  //Put your DataWeave mappings files here
|   |   |   └── MyModule.dwl   //Put your DataWeave modules files here
│   │   └── resources
│   └── test
│       ├── dw
│       │   └── MyModuleTest.dwl  //Unit tests for the functions in your module
│       │   └── MyMappingTest.dwl //Integration test for your mapping using the saved scenarios
│       ├── resources
│       │       └── MyMapping //Contains the different scenarios for your mapping
│       │           ├── default
│       │           │   └── inputs
│       │           │       └── payload.json //[Sample data](#sample-data) for your mappings
│       │           └── SavedTestScenario //Scenario used for the integration test
├── pom.xml
```
The scenarios are inside test/resources. Added some comments but I'm not sure if this confuses the reader with the concept of scenarios that at this point it is not yet explained.